src/pages/ImportData.js

```diff
@@ -63,15 +91,66 @@ const ImportData = () => {
     div({ role: 'main', style: styles.container }, [
       div({ style: styles.card }, [
         div({ style: styles.title }, ['Importing Data']),
-        div({ style: { fontSize: 16 } }, ['From: ', new URL(url).hostname])
+        div({ style: { fontSize: 16 } }, ['From: ', new URL(url).hostname]),
+        div({ style: { marginTop: '1rem' } }, [
+          'The dataset(s) you just chose to import to Terra will be made available to you within a workspace of your choice where you can then perform analysis.'
+        ])
       ]),
       div({ style: { ...styles.card, marginLeft: '2rem' } }, [
-        div({ style: styles.title }, ['Destination Workspace']),
-        h(WorkspaceImporter, {
-          authorizationDomain: ad,
-          selectedWorkspaceId: wid,
-          onImport
-        }),
+        Utils.switchCase(mode,
+          ['existing', () => {
+            return h(Fragment, [
+              div({ style: styles.title }, ['Start with an existing workspace']),
+              h(WorkspaceSelector, {
+                workspaces: _.filter(ws => {
+                  return Utils.canWrite(ws.accessLevel) &&
+                    (!ad || _.some({ membersGroupName: ad }, ws.workspace.authorizationDomain))
+                }, workspaces),
+                value: selectedWorkspaceId,
+                onChange: setSelectedWorkspaceId
+              }),
+              div({ style: { display: 'flex', alignItems: 'center', marginTop: '1rem' } }, [
+                h(ButtonPrimary, {
+                  disabled: !selectedWorkspace,
+                  onClick: () => onImport(selectedWorkspace.workspace)
+                }, ['Import']),
+                h(ButtonSecondary, { style: { marginLeft: '1rem' }, onClick: () => setMode() }, ['Back'])
+              ])
+            ])
+          }],
+          [Utils.DEFAULT, () => {
+            return h(Fragment, [
+              div({ style: styles.title }, ['Destination Workspace']),
+              div({ style: { marginTop: '0.5rem' } }, ['Choose the option below that best suits your needs.']),
+              div({ style: { marginTop: '0.5rem' } }, ['Note that the import process may take some time after you are redirected into your destination workspace.']),
+              h(ChoiceButton, {
+                style: { marginTop: '1rem' },
+                onClick: () => setMode('existing'),
+                iconName: 'folder-open',
+                title: 'Start with an existing workspace',
+                detail: 'Select one of your workspaces'
+              }),
+              h(ChoiceButton, {
+                style: { marginTop: '1rem' },
+                onClick: () => setIsCreateOpen(true),
```

Comment:
I think you could get away with folding this state into `mode`, more like it was previously. However, that would also require changing the `switchCase` to `cond`, and I can see that having an explicit `isCreateOpen` flag makes the state more closely mirror the visible UI state.